File: packages/python-sdk/src/blindfold/regex/detectors/sk/dic.py
"""Slovak DIC (tax identification number) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import sk_dic_valid


@register_region("sk")
class DicDetector(RegexDetector):
    entity_type = EntityType.SK_DIC
    score = 0.85
    context_keywords = [
        "dic", "di\u010d", "danove identifikacne cislo",
        "da\u0148ov\u00e9 identifika\u010dn\u00e9 \u010d\u00edslo", "vat", "tax id",
    ]
    context_required = False
    context_window = 50

    # SK prefix + 10 digits
    pattern = re.compile(
        r"\bSK\d{10}\b",
        re.IGNORECASE,
    )
    validator = staticmethod(sk_dic_valid)

File: packages/js-sdk/src/regex/detectors/sk/dic.ts
// Slovak DIC (tax identification number) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { skDicValid } from '../../validators'

class SkDicDetector extends RegexDetector {
  entityType = EntityType.SK_DIC
  score = 0.85
  contextKeywords = ['dic', 'dič', 'danove identifikacne cislo', 'daňové identifikačné číslo', 'vat', 'tax id']
  contextRequired = false

  pattern = /\bSK\d{10}\b/gi
  validator = skDicValid
}

registerRegion('sk', SkDicDetector)

export { SkDicDetector }
